dhall/src/Dhall/Parser.hs

```diff
@@ -108,10 +121,10 @@ exprAndHeaderFromText
     :: String -- ^ User-friendly name describing the input expression,
               --   used in parsing error messages
     -> Text   -- ^ Input expression to parse
-    -> Either ParseError (Text, Expr Src Import)
+    -> Either ParseError (Header, Expr Src Import)
 exprAndHeaderFromText delta text = case result of
     Left errInfo   -> Left (ParseError { unwrap = errInfo, input = text })
-    Right (txt, r) -> Right (stripHeader txt, r)
+    Right (txt, r) -> Right (createHeader txt, r)
```

Comment:
IMHO it would be better to leave the whitespace stripping to the pretty-printer. But that can be changed later on.

Reply:
That would probably make more sense, I just left the whitespace stripping in the parser as it was. I'm also not too sure why it's done like this.

I think the concept of a header comment might make sense if we want to exploit it for documentation (maybe with markdown) in the future, but it needs to be a lot more strict. Currently it allows for multiple comments at once mixing single line and multiline.